spec/lib/booking_requests/booking_requests_spec.rb

```diff
@@ -2,8 +2,16 @@
   let(:api) { instance_double(BookingRequests::Api) }
 
   describe '.slots' do
-    before do
+    around do |example|
+      previous_api = BookingRequests.api
       BookingRequests.api = api
+
+      example.run
+
+      BookingRequests.api = previous_api
```

Comment:
I've only seen it in cukes and they're extracted into support helpers that are pulled in, based on tags.